x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -265,29 +276,157 @@ func (k Keeper) SettlePendingClaims(ctx sdk.Context) (
+		if err = k.slashSupplierStake(ctx, supplierOperatorAddress, slashingCount); err != nil {
 			logger.Error(fmt.Sprintf("error slashing supplier %s: %s", supplierOperatorAddress, err))
-			return settledResult, expiredResult, err
+			return settledResults, expiredResults, err
 		}
 	}
 
 	logger.Info(fmt.Sprintf(
 		"settled %d and expired %d claims at block height %d",
-		settledResult.NumClaims,
-		expiredResult.NumClaims,
+		settledResults.GetNumClaims(),
+		expiredResults.GetNumClaims(),
 		blockHeight,
 	))
 
-	return settledResult, expiredResult, nil
+	return settledResults, expiredResults, nil
+}
+
+// ExecutePendingResults executes all pending mint, burn, and transfer operations.
+func (k Keeper) ExecutePendingResults(ctx cosmostypes.Context, results tlm.PendingSettlementResults) (errs error) {
+	logger := k.logger.With("method", "ExecutePendingResults")
+
+	for _, result := range results {
+		logger = logger.With("session_id", result.GetSessionId())
+		logger.Info("begin executing pending results")
+
+		logger.Info("begin executing pending mints")
+		errs = k.executePendingMints(ctx, result.Mints)
+		logErrors(logger, errs)
+
+		logger.Info("done executing pending mints")
+
+		logger.Info("begin executing pending burns")
+		if err := k.executePendingBurns(ctx, result.Burns); err != nil {
+			logErrors(logger, err)
+			errs = errors.Join(errs, err)
+		}
+		logger.Info("done executing pending burns")
+
+		logger.Info("begin executing pending module to module transfers")
+		if err := k.executePendingModToModTransfers(ctx, result.ModToModTransfers); err != nil {
+			logErrors(logger, err)
+			errs = errors.Join(errs, err)
+		}
+		logger.Info("done executing pending module to module transfers")
+
+		logger.Info("begin executing pending module to account transfers")
+		if err := k.executePendingModToAcctTransfers(ctx, result.ModToAcctTransfers); err != nil {
+			logErrors(logger, err)
+			errs = errors.Join(errs, err)
+		}
+		logger.Info("done executing pending module to account transfers")
+
+		logger.Info("done executing pending results")
+	}
+	return errs
+}
+
+// logErrors recursively unwraps and sequentially log all joined errors in errs.
+func logErrors(logger cosmoslog.Logger, errs error) {
+	if pokterrors.Split(errs) == nil {
+		return
+	}
+
+	for _, err := range pokterrors.Split(errs) {
+		logger.Error(err.Error())
+	}
+}
+
+// executePendingMints executes all pending mint operations.
+func (k Keeper) executePendingMints(ctx cosmostypes.Context, mints []tlm.MintBurn) (errs error) {
+	for _, mint := range mints {
+		if err := k.bankKeeper.MintCoins(ctx, mint.DestinationModule, cosmostypes.NewCoins(mint.Coin)); err != nil {
+			err = tokenomicstypes.ErrTokenomicsModuleMint.Wrapf(
+				"destination module %q minting %s: %+v", mint.DestinationModule, mint.Coin, err,
+			)
+			errs = errors.Join(errs, err)
+		}
+	}
+	return errs
+}
+
+// executePendingBurns executes all pending burn operations.
```

Comment:
#PUC and function that it's always a module whose coins we're burning.

Reply:
```go
// executePendingModuleBurns executes all pending burn operations.
// DEV_NOTE: Mint and burn operations are ONLY applicable to module accounts.
func (k Keeper) executePendingModuleBurns(ctx cosmostypes.Context, burns []tokenomicstypes.MintBurnOp) error {
```